app/bundles/AssetBundle/EventListener/DetermineWinnerSubscriber.php

```diff
@@ -1,37 +1,66 @@
 <?php
 
 /*
- * @copyright   2014 Mautic Contributors. All rights reserved
+ * @copyright   2019 Mautic Contributors. All rights reserved
  * @author      Mautic
  *
  * @link        http://mautic.org
  *
  * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
  */
 
-namespace Mautic\AssetBundle\Helper;
+namespace Mautic\AssetBundle\EventListener;
 
-use Mautic\CoreBundle\Factory\MauticFactory;
+use Doctrine\ORM\EntityManager;
+use Mautic\AssetBundle\AssetEvents;
+use Mautic\CoreBundle\Event\DetermineWinnerEvent;
 use Mautic\EmailBundle\Entity\Email;
-use Mautic\PageBundle\Entity\Page;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Translation\TranslatorInterface;
 
-/**
- * Class AbTestHelper.
- */
-class AbTestHelper
+class DetermineWinnerSubscriber implements EventSubscriberInterface
 {
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param EntityManager       $em
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(EntityManager $em, TranslatorInterface $translator)
+    {
+        $this->em         = $em;
+        $this->translator = $translator;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            AssetEvents::ON_DETERMINE_DOWNLOAD_RATE_WINNER => ['onDetermineDownloadRateWinner', 0],
+        ];
+    }
+
     /**
      * Determines the winner of A/B test based on number of asset downloads.
      *
-     * @param MauticFactory $factory
-     * @param Page          $parent
-     * @param               $children
-     *
-     * @return array
+     * @param DetermineWinnerEvent $event
      */
-    public static function determineDownloadWinner($factory, $parent, $children)
+    public function onDetermineDownloadRateWinner(DetermineWinnerEvent $event)
     {
-        $repo = $factory->getEntityManager()->getRepository('MauticAssetBundle:Download');
+        $repo       = $this->em->getRepository('MauticAssetBundle:Download');
```

Comment:
```suggestion
        $repo       = $this->em->getRepository(Download::class);
```
+ add the use statement of the Download entity.

Reply:
Done.